eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/ImmutableBagFactory.java

```diff
@@ -47,8 +49,111 @@
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2);
+    }
+
+    /**
+     * Same as {@link #withOccurrences(Object, int, Object, int, Object, int)}.
+     *
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2, element3, occurrence3);
+    }
+
+    /**
+     * Same as {@link #withOccurrences(Object, int, Object, int, Object, int, Object, int)}.
+     *
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3, T element4, int occurrence4)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2, element3, occurrence3, element4, occurrence4);
+    }
+
+    /**
+     * Same as {@link #withOccurrences(ObjectIntPair[])}
+     *
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> ofOccurrences(ObjectIntPair<T>... elementsWithOccurrences)
+    {
+        return this.withOccurrences(elementsWithOccurrences);
+    }
+
     <T> ImmutableBag<T> with(T... elements);
 
+    /**
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> withOccurrences(T element, int occurrence)
+    {
+        MutableBag<T> mutableBag = Bags.mutable.withOccurrences(element, occurrence);
+        return mutableBag.toImmutable();
+    }
+
+    /**
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2)
+    {
+        MutableBag<T> mutableBag = Bags.mutable.withOccurrences(
+                element1, occurrence1,
+                element2, occurrence2);
+        return mutableBag.toImmutable();
+    }
+
+    /**
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3)
+    {
+        MutableBag<T> mutableBag = Bags.mutable.withOccurrences(
+                element1, occurrence1,
+                element2, occurrence2,
+                element3, occurrence3);
+        return mutableBag.toImmutable();
+    }
+
+    /**
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3, T element4, int occurrence4)
+    {
+        MutableBag<T> mutableBag = Bags.mutable.withOccurrences(
+                element1, occurrence1,
+                element2, occurrence2,
+                element3, occurrence3,
+                element4, occurrence4);
+        return mutableBag.toImmutable();
+    }
+
+    /**
+     * @since 11.0
+     */
+    default <T> ImmutableBag<T> withOccurrences(ObjectIntPair<T>... elementsWithOccurrences)
+    {
+        return Bags.mutable.withOccurrences(elementsWithOccurrences).toImmutable();
+    }
```

Comment:
The "since 11.0" is a guess. We could potentially release this in 10.3.0. Just an observation, this is fine.

Reply:
It should be 10.3.0. I am planning to have a minor release before a major release

Reply:
Changed to `@since 10.3`.